### PR TITLE
Implement carrier service common for shipping modules

### PR DESCRIPTION
**Pull request by priyankarani**

### CODE REVIEW

File: carrier.py [hunk 1]
```diff
@@ -41,3 +42,47 @@ def get_rates(self):
...
+
+    active = fields.Boolean('Active', select=True)
+    name = fields.Char('Name', required=True, select=True, readonly=True)
+    value = fields.Char('Value', required=True, select=True, readonly=True)
+    method_type = fields.Selection(
+        'get_method_type', 'Type', select=True, readonly=True,
+        states={'required': Bool(Eval('is_method_type_required'))},
+        depends=['is_method_type_required']
+    )
+    display_name = fields.Char('Display Name', select=True)
```
Comment (prakashpp) [inline carrier.py]: not needed

File: carrier.py [hunk 2]
```diff
@@ -41,3 +42,47 @@ def get_rates(self):
...
+    source = fields.Selection(
```
Comment (prakashpp) [inline carrier.py]: why not a many2one carrier?

File: carrier.py [hunk 3]
```diff
@@ -41,3 +42,47 @@ def get_rates(self):
...
+
+
+class CarrierService(ModelSQL, ModelView):
+    "Carrier Service"
+    __name__ = 'carrier.service'
+
+    active = fields.Boolean('Active', select=True)
+    name = fields.Char('Name', required=True, select=True, readonly=True)
+    value = fields.Char('Value', required=True, select=True, readonly=True)
+    method_type = fields.Selection(
```
Comment (prakashpp) [inline carrier.py]: why not char?

File: __init__.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ def register():
     Pool.register(
         PartyConfiguration,
         Carrier,
+        CarrierService,
```
Comment (sharoonthomas) [inline __init__.py]: just Service

File: carrier.py [hunk 5]
```diff
@@ -41,3 +42,47 @@ def get_rates(self):
...
         return []  # pragma: no cover
+
+
+class CarrierService(ModelSQL, ModelView):
+    "Carrier Service"
+    __name__ = 'carrier.service'
+
+    active = fields.Boolean('Active', select=True)
+    name = fields.Char('Name', required=True, select=True, readonly=True)
+    value = fields.Char('Value', required=True, select=True, readonly=True)
```
Comment (sharoonthomas) [inline carrier.py]: call it code

File: carrier.py [hunk 6]
```diff
@@ -41,3 +42,47 @@ def get_rates(self):
...
         # method in sale instead
         return []  # pragma: no cover
+
+
+class CarrierService(ModelSQL, ModelView):
+    "Carrier Service"
+    __name__ = 'carrier.service'
+
+    active = fields.Boolean('Active', select=True)
+    name = fields.Char('Name', required=True, select=True, readonly=True)
```
Comment (sharoonthomas) [inline carrier.py]: this should not be readonly

**Pull request closed without merging**